fastq_filtrator.py

```diff
@@ -0,0 +1,241 @@
+# Here is fastq filtrator
+
+
+# GC-filtering
+def filter_gc(input_list, gc_bounds, save_filtered):
+    input_list_1 = list()
+    output_list_failed_1 = list()
+    for n in range(0, len(input_list)):
+        if n % 4 == 1:
+            gc = 100*(input_list[n].count('G') + input_list[n].count('С')) / len(input_list[n])
+            if float(gc_bounds[0]) <= gc <= float(gc_bounds[1]):
+                input_list_1.append(input_list[n - 1])
+                input_list_1.append(input_list[n])
+                input_list_1.append(input_list[n + 1])
+                input_list_1.append(input_list[n + 2])
+            else:
+                if save_filtered is True:
+                    output_list_failed_1.append(input_list[n - 1])
+                    output_list_failed_1.append(input_list[n])
+                    output_list_failed_1.append(input_list[n + 1])
+                    output_list_failed_1.append(input_list[n + 2])
+    return(input_list_1, output_list_failed_1)
+
+
+# length filtering
+def filter_length(input_list, length_bounds, save_filtered, gc_bounds):
+    input_list, _ = filter_gc(input_list, gc_bounds, save_filtered)
+    input_list_2 = list()
+    output_list_failed_2 = list()
+    _, output_from_1 = filter_gc(input_list, gc_bounds, save_filtered)
+    if input_list == []:
+        return(input_list, output_list_failed_2 + output_from_1)
+    for n in range(0, len(input_list)):
+        if n % 4 == 1:
+            length = len(input_list[n])
+            if float(length_bounds[0]) <= length <= float(length_bounds[1]):
+                input_list_2.append(input_list[n - 1])
+                input_list_2.append(input_list[n])
+                input_list_2.append(input_list[n + 1])
+                input_list_2.append(input_list[n + 2])
+            else:
+                if save_filtered is True:
+                    output_list_failed_2.append(input_list[n - 1])
+                    output_list_failed_2.append(input_list[n])
+                    output_list_failed_2.append(input_list[n + 1])
+                    output_list_failed_2.append(input_list[n + 2])
+    return(input_list_2, output_list_failed_2 + output_from_1)
+
+
+# quality filtering
+def filter_quality(input_list, quality_threshold, save_filtered, length_bounds, gc_bounds):
+    input_list_final, _ = filter_length(input_list, length_bounds, save_filtered, gc_bounds)

```

Comment:
Здесь и дальше ты вызываешь одну и ту же функцию с одними и теми же параметрами, сохраняя разные переменные. Хотя вроде бы всё можно сделать за 1 вызов
```suggestion
    input_list_final, output_from_2 = filter_length(input_list, length_bounds, save_filtered, gc_bounds)

```